jaeger-core/src/test/java/io/jaegertracing/internal/propagation/TraceContextCodecTest.java

```diff
@@ -160,10 +164,31 @@ public void testEmptyTraceStateNotPropagated() {
     assertEquals(EXAMPLE_TRACE_PARENT, injectCarrier.get(TRACE_PARENT));
   }
 
+  @Test
+  public void testDebugIdWithoutTraceHeader() {
+    Map<String, String> extractCarrier = new HashMap<>();
+    TextMapAdapter textMap = new TextMapAdapter(extractCarrier);
+    textMap.put(Constants.DEBUG_ID_HEADER_KEY, EXAMPLE_DEBUG_ID);
+    JaegerSpanContext spanContext = traceContextCodec.extract(textMap);
+    assertEquals(EXAMPLE_DEBUG_ID, spanContext.getDebugId());
```

Comment:
you can mock `objectFactory` and verify that the debug ID is passed this way, instead of exposing a new method on the context. Or you can create a tracer with this codec, start a span, and verify that span has the tag recorded.